src/object/cli_obj.c

```diff
@@ -5415,6 +5415,14 @@ dc_obj_update(tse_task_t *task, struct dtx_epoch *epoch, uint32_t map_ver,
 		tgt_bitmap = obj_auxi->reasb_req.tgt_bitmap;
 	}
 
+	/* The data might be needed to forwarded to other targets (or not forwarded anymore)
+	 * after pool map refreshed, especially during online extending or reintegration,
+	 * which needs to be binded or unbinded.
+	 * So let's free the existent bulk, and recreate the bulk later.
+	 */
+	if (obj_auxi->io_retry && obj_auxi->bulks != NULL)
+		obj_bulk_fini(obj_auxi);
```

Comment:
just confirm for this bulk change, why need to recreate the bulk handle?
note that for crt_bulk_bind() is it to bind local context address to the bulk handle (not remote target address), so not sure why need to recreate (for rebind?) here?

Reply:
yes, this is for rebinding.  for osa online case.  it is possible that the initial I/O will not need bind the bulk because it does not forward, but retry will refresh the pool map, then need forward (need bulk the bind). 

On the other hand,  the initial I/O might need forward, but then retry might not need forward. 

So we might just delete the bulk and recreate and re-bind the bulk as needed.